### PR TITLE
[RFC] Fix stall before starting tui when something is output in Windows

### LINKED ISSUE #7967: Windows/TUI: Printing to stdout during initialization prevents starting normally
- `nvim --version`: NVIM v0.2.2 (also with very recent 0.2.3-580-gec62f925a)
- Vim (version: ) behaves differently? No
- Operating system/version: Windows 10 Pro 1709
- Terminal name/version: cmd.exe
- `$TERM`: empty, cygwin (when running cmd.exe under cmder)

### Steps to reproduce

```
nvim -u init.test.vim
```

with `init.test.vim` being:

```
echom "Test echom"
```

### Actual behaviour

The following gets printed to the terminal:

```
Test echom
Press ENTER or type command to continue
```

And nothing else happens no matter what keys you press. Only `ctrl-c` gets you out to the terminal, and then everything you typed gets printed.

This seems to happen when anything prints messages during initialization.

I am able to start normally even with a complex init file only if no messages get printed.

### Expected behaviour

Display the message and start normally.

### PR DESCRIPTION
**Pull request by erw7**
fixes #7967

### CODE REVIEW

**Comment by erw7** [comment]
I don't know if it's really all right, but is it better to fix `stream_init()` like https://github.com/neovim/neovim/commit/a48ea6806dac6f27accc8ca19628538111ea98f2?

**Comment by justinmk** [comment]
> I don't know if it's really all right, but is it better to fix `stream_init()` like [6589f6f](https://github.com/neovim/neovim/commit/6589f6f418fed08fd9e9f1d063989543e9324c22)?

That seems slightly better to me, if this is something that's always required for streams on Windows.

But for #7967 I think the best approach is to simply not print anything until startup is ready, as  @bfredl  [mentioned](https://github.com/neovim/neovim/issues/7967#issuecomment-433875000). Alternatively, collect messages in a list which is later printed.

**Comment by erw7** [comment]
I do not understand exactly what it does, but I implemented it referring to the embedded case. The behavior is different from vim(A message is displayed after the screen is cleared, but on vim a message is displayed before the screen is cleared), but it seems to be working. This closes because I created #9829.

File: src/nvim/tui/input.c [hunk 1]
```diff
@@ -49,14 +49,9 @@ void tinput_init(TermInput *input, Loop *loop)
...
-#ifdef WIN32
-  uv_tty_init(&loop->uv, &input->tty_in, 0, 1);
-  uv_tty_set_mode(&input->tty_in, UV_TTY_MODE_RAW);
-  rstream_init_stream(&input->read_stream,
-                      (uv_stream_t *)&input->tty_in,
-                      0xfff);
-#else
   rstream_init_fd(loop, &input->read_stream, input->in_fd, 0xfff);
+#ifdef WIN32
+  uv_tty_set_mode(&input->read_stream.uv.tty, UV_TTY_MODE_RAW);
```
Comment (justinmk) [inline src/nvim/tui/input.c]: why not do this in `stream_init` ?

Comment (erw7) [reply src/nvim/tui/input.c]: All current calls need to do `uv_tty_set_mode(..., ..., UV_TTY_MODE_RAW)`, so I think that's fine. There may be calls that do not need it, so I did not.

**Pull request closed without merging**